src/types.ts

```diff
@@ -6,16 +6,16 @@
  * The success case of @link AccountsRequestResult.
  */
 export type SuccessAccountsRequestResult<T> = {
-  error: null | undefined
-  data: T
+  data: T | null
+  error: undefined
```

Comment:
Are we sure the `error: undefined` stays here?

Reply:
Yup, its for Typescript to know when error is `undefined`, it'll be  a `SuccessAccountsRequestResult`.
```ts
export type AccountsRequestResult<T> =
  | SuccessAccountsRequestResult<T>
  | FailureAccountsRequestResult
```